### ISSUE TITLE
Collaborative Password Manager

### ISSUE THREAD

**Issue by silky** [silky/ideas#365]
say i want to make a password for a system you will control, but we both need access to

maybe i can have my program generate part of it; you're program generate part, then combine them both on our independent computers, without the entire password leaving either of them

could build this on top of the public keys on github somehow; so i just pick the github user i'm going to share a password with

could clearly do this immediately by encrypting it with their public key, or something.

but maybe something richer can be done

**Comment by dcousens** [comment]
https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing

**Comment by dcousens** [comment]
> maybe i can have my program generate part of it; you're program generate part, then combine them both on our independent computers, without the entire password leaving either of them

Sounds like https://en.wikipedia.org/wiki/Diffie-Hellman_key_exchange
![](https://upload.wikimedia.org/wikipedia/commons/4/46/Diffie-Hellman_Key_Exchange.svg)

**Comment by silky** [comment]
yeah, there's that, but i also want something on top of that that is somehow quirkier


something like the secret sharing. but i can't quite describe it well
